### PR TITLE
[Docs] Telemetry page

### PR DESCRIPTION
**Pull request by martintmk**
### Details on the issue fix or feature implementation

Finish the docs for the telemetry by shuffling the content between existing docs.

Contributes to #1091 

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: docs/telemetry.md [hunk 1]
```diff
@@ -1,3 +1,156 @@
...
+
+        // Configure telemetry listeners
+        options.TelemetryListeners.Add(new MyTelemetryListener());
+    });
+```
+<!-- endSnippet -->
+
+## Metrics
+
+The emitted metrics are emitted under the `Polly` meter name. The subsequent sections provide insights into the metrics produced by Polly. Please note that any custom enriched dimensions are not depicted in the following tables.
```
Comment (martincostello) [inline docs/telemetry.md:78]:
```suggestion
The metrics are emitted under the `Polly` meter name. The subsequent sections provide insights into the metrics produced by Polly. Please note that any custom enriched dimensions are not depicted in the following tables.
```

File: docs/telemetry.md [hunk 2]
```diff
@@ -1,3 +1,156 @@
...
+
+Dimensions:
+
+|Name|Description|
+|---| ---|
+|`event-name`| The name of the emitted event.|
+|`event-severity`| The severity of the event (`Debug`, `Information`, `Warning`, `Error`, `Critical`).|
+|`pipeline-name`| The name of the pipeline corresponding to the resilience pipeline.|
+|`pipeline-instance`| The instance name of the pipeline corresponding to the resilience pipeline.|
+|`strategy-name`| The name of the strategy generating this event.|
```
Comment (martincostello) [inline docs/telemetry.md:93]: Maybe note that this isn't always set like you were telling me #1546?

File: docs/telemetry.md [hunk 3]
```diff
@@ -1,3 +1,156 @@
...
+|`operation-key`| The operation key associated with the call site. |
+|`exception-name`| The full name of the exception assigned to the execution result (`System.InvalidOperationException`). |
+|`attempt-number`| The execution attempt number, starting at 0 (0, 1, 2). |
```
Comment (martincostello) [inline docs/telemetry.md:114]:
```suggestion
|`attempt-number`| The execution attempt number, starting at 0 (0, 1, 2, etc.). |
```

File: docs/telemetry.md [hunk 4]
```diff
@@ -1,3 +1,156 @@
...
+|`attempt-handled`| Indicates if the execution outcome was handled. A handled outcome indicates execution failure and the need for retry (`true`, `false`). |
+
+### pipeline-execution-duration
+
+- Type: *Histogram*
+- Unit: *milliseconds*
+- Description: Measures the duration and results of resilience pipelines.
```
Comment (martincostello) [inline docs/telemetry.md:121]: It doesn't contain the result anymore does it?

Comment (martintmk) [reply docs/telemetry.md:121]: True.

**Pull request merged**